services/wallet/router/routererrors/errors.go

```diff
@@ -0,0 +1,73 @@
+package routererrors
+
+import (
+	"encoding/json"
+)
+
+// ErrorCode represents a specific error code.
+type ErrorCode string
+
+// ErrorResponse represents an error response structure.
+type ErrorResponse struct {
+	Code    ErrorCode `json:"code"`
+	Details string    `json:"details,omitempty"`
+}
+
+// Error implements the error interface for ErrorResponse.
+func (e *ErrorResponse) Error() string {
+	errorJSON, _ := json.Marshal(e)
+	return string(errorJSON)
+}
+
+// CreateErrorResponseFromError creates an ErrorResponse from a generic error.
+func CreateErrorResponseFromError(err error) error {
+	if err == nil {
+		return nil
+	}
+	if errResp, ok := err.(*ErrorResponse); ok {
+		return errResp
+	}
+	return &ErrorResponse{
+		Code:    "0",
+		Details: err.Error(),
+	}
+}
+
+var (
+	ErrUsernameAndPubKeyRequiredForENSRegister   = &ErrorResponse{Code: ErrorCode("R-001"), Details: "username and public key are required for ENSRegister"}
+	ErrOnlySTTSupportedForENSRegisterOnTestnet   = &ErrorResponse{Code: ErrorCode("R-002"), Details: "only STT is supported for ENSRegister on testnet"}
+	ErrOnlySTTSupportedForENSReleaseOnTestnet    = &ErrorResponse{Code: ErrorCode("R-003"), Details: "only STT is supported for ENSRelease on testnet"}
+	ErrUsernameRequiredForENSRelease             = &ErrorResponse{Code: ErrorCode("R-004"), Details: "username is required for ENSRelease"}
+	ErrUsernameAndPubKeyRequiredForENSSetPubKey  = &ErrorResponse{Code: ErrorCode("R-005"), Details: "username and public key are required for ENSSetPubKey"}
+	ErrPackIDRequiredForStickersBuy              = &ErrorResponse{Code: ErrorCode("R-006"), Details: "packID is required for StickersBuy"}
+	ErrToTokenIDRequiredForSwap                  = &ErrorResponse{Code: ErrorCode("R-007"), Details: "toTokenID is required for Swap"}
```

Comment:
Very cool implementation 😉 , I wonder if these errors now could be within the `router` package.